src/parser/hl7Component.ts

```diff
@@ -0,0 +1,15 @@
+import { HL7SubComponent } from './hl7SubComponent';
+import { Parser } from './parse';
+
+export class HL7Component {
+    HL7SubComponents: HL7SubComponent[] = [];
+    Index: number;
+    Value: string;
+    HasSubComponents: boolean;
+    isHighlighted: boolean;
+
+    constructor(hl7Component: string, hl7ComponentIndex: number) {
+        this.Value = hl7Component;
+        this.Index = hl7ComponentIndex;
+    }
+}
```

Comment:
In TypeScript, you can do this:

`
constructor(private Value: string, private Index: number) { }
`

Then you don't have to declare them as class variables.  So you can do that and remove lines 6 and 7, and it will work like you want it to.  Also, generally when you write TS/JS, your variables (even class variables) are start with lower-case letters.